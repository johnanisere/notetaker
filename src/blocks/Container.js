import styled, { css } from "styled-components";

export default styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  max-width: 1270px;
  ${({ center }) =>
    center &&
    css`
      padding-top: 100px;
      justify-content: center;
      align-items: center;
      max-width: 600px;
    `}
  ${({ wrap }) =>
    wrap &&
    css`
      flex-direction: row;
      flex-wrap: wrap;
      padding-left: 20px;
      padding-top: 30px;
      @media (max-width: 1316px) {
        max-width: 838px;
      }
      @media (max-width: 897px) {
        max-width: 420px;
      }
    `}
`;
