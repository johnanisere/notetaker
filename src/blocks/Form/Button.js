import styled from "styled-components";

import Button from "../../elements/Button";

export default styled(Button)`
  background: -webkit-linear-gradient(top right, #33bef6, #26aef4);
  color: #ffffff;
  font-size: 15px;
  margin-top: 15px;
  border-radius: 5px;
  cursor: pointer;
  border-color: #33bef6;
  padding: 10px 25px;
  font-weight: 500;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
  transition: all 0.15s ease ease-in-out;
  &:hover {
    transform: translateY(-1px);
    box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
  }
`;
