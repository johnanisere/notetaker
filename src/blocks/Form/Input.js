import styled from "styled-components";

import Input from "../../elements/Input";

export default styled(Input)`
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background: rgba(0, 0, 0, 0.03);
`;
