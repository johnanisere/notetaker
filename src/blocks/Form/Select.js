import styled from "styled-components";

import Select from "../../elements/Select";

export default styled(Select)`
  padding: 12px;
  font-size: 16px;
  font-weight: 500;
  background-color: rgba(0, 0, 0, 0.03);
  appearance: none;
  border: 1px solid #d4d4d4;
  padding: 12px;
  box-sizing: border-box;
  margin-top: 8px;
  border-radius: 4px;
`;
