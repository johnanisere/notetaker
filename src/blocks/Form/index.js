import styled from "styled-components";

import Input from "./Input";
import Select from "./Select";
import Button from "./Button";
import Label from "../../elements/Label";

const Form = styled.form``;

Form.Input = Input;
Form.Label = Label;
Form.Select = Select;
Form.Button = Button;

export default Form;
