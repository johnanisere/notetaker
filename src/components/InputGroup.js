import React from "react";
import Form from "../blocks/Form";
import Select from "./Select";

const Group = ({ title, value, name, onChange, type, list = [] }) => (
  <Form.Label>
    {title}
    <br />
    {type === "select" ? (
      <Select value={value} name={name} onChange={onChange} list={list} />
    ) : (
      <Form.Input value={value} name={name} type={type} onChange={onChange} />
    )}
  </Form.Label>
);

export default Group;
