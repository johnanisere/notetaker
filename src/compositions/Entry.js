import React from "react";
import { colors } from "./constants";
import Form from "../blocks/Form";
import Container from "../blocks/Container";
import InputGroup from "../components/InputGroup";

const App = ({ note, color, onSave, onHandleChange }) => (
  <Container center>
    <InputGroup
      type="text"
      name="note"
      value={note}
      title="Note Text"
      onChange={onHandleChange}
    />
    <InputGroup
      name="color"
      type="select"
      list={colors}
      value={color}
      title="Note Color"
      onChange={onHandleChange}
    />
    <Form.Button onClick={onSave}>ADD NOTE</Form.Button>
  </Container>
);

export default App;
