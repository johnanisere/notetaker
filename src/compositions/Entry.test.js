import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Entry from "./Entry";
import Form from "../blocks/Form";
import Container from "../blocks/Container";
import InputGroup from "../components/InputGroup";

Enzyme.configure({ adapter: new Adapter() });

it("Entry component exists", () => {
  const wrapper = shallow(<Entry />);
  expect(wrapper.exists()).toBe(true);
});

it("Entry component renders", () => {
  const wrapper = shallow(<Entry />);

  expect(wrapper).toMatchSnapshot();
});

it(`Cards should contain "Container","InputGroup" and "Button"`, () => {
  const mockCallBack = jest.fn();
  const props = {
    note: "Hey",
    color: "Red",
    onSave: mockCallBack,
    onHandleChange: mockCallBack
  };
  expect.assertions(4);
  const wrapper = shallow(<Entry {...props} />);
  expect(wrapper.find(Container)).toHaveLength(1);
  expect(wrapper.find(InputGroup)).toHaveLength(2);
  expect(wrapper.find(Form.Button)).toHaveLength(1);
  wrapper.find(Form.Button).simulate("click");
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
