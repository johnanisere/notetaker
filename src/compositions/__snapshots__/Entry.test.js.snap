// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Entry />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "center": true,
      "children": Array [
        <Group
          name="note"
          title="Note Text"
          type="text"
        />,
        <Group
          list={
            Array [
              "Red",
              "Blue",
              "Gold",
              "Maroon",
              "Orange",
              "Pink",
              "Plum",
              "Brown",
            ]
          }
          name="color"
          title="Note Color"
          type="select"
        />,
        <ForwardRef>
          ADD NOTE
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "note",
          "onChange": undefined,
          "title": "Note Text",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "list": Array [
            "Red",
            "Blue",
            "Gold",
            "Maroon",
            "Orange",
            "Pink",
            "Plum",
            "Brown",
          ],
          "name": "color",
          "onChange": undefined,
          "title": "Note Color",
          "type": "select",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "ADD NOTE",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "ADD NOTE",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  background: -webkit-linear-gradient(top right, #33bef6, #26aef4);
  color: #ffffff;
  font-size: 15px;
  margin-top: 15px;
  border-radius: 5px;
  cursor: pointer;
  border-color: #33bef6;
  padding: 10px 25px;
  font-weight: 500;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
  transition: all 0.15s ease ease-in-out;
  &:hover {
    transform: translateY(-1px);
    box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
  }
",
            ],
          },
          "displayName": "Styled(styled.button)",
          "foldedComponentIds": Array [
            "sc-ifAKCX",
          ],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htoDjs",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  max-width: 1270px;
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htoDjs",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "center": true,
        "children": Array [
          <Group
            name="note"
            title="Note Text"
            type="text"
          />,
          <Group
            list={
              Array [
                "Red",
                "Blue",
                "Gold",
                "Maroon",
                "Orange",
                "Pink",
                "Plum",
                "Brown",
              ]
            }
            name="color"
            title="Note Color"
            type="select"
          />,
          <ForwardRef>
            ADD NOTE
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "note",
            "onChange": undefined,
            "title": "Note Text",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "list": Array [
              "Red",
              "Blue",
              "Gold",
              "Maroon",
              "Orange",
              "Pink",
              "Plum",
              "Brown",
            ],
            "name": "color",
            "onChange": undefined,
            "title": "Note Color",
            "type": "select",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "ADD NOTE",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "ADD NOTE",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  background: -webkit-linear-gradient(top right, #33bef6, #26aef4);
  color: #ffffff;
  font-size: 15px;
  margin-top: 15px;
  border-radius: 5px;
  cursor: pointer;
  border-color: #33bef6;
  padding: 10px 25px;
  font-weight: 500;
  box-shadow: 0 4px 6px rgba(50, 50, 93, 0.11), 0 1px 3px rgba(0, 0, 0, 0.08);
  transition: all 0.15s ease ease-in-out;
  &:hover {
    transform: translateY(-1px);
    box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
  }
",
              ],
            },
            "displayName": "Styled(styled.button)",
            "foldedComponentIds": Array [
              "sc-ifAKCX",
            ],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htoDjs",
          "isStatic": false,
          "rules": Array [
            "
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
  max-width: 1270px;
  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htoDjs",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
